src/virtual-source-buffer.js

```diff
@@ -66,6 +83,42 @@ export const gopsSafeToAlignWith = (buffer, player, mapping) => {
   return buffer.slice(i);
 };
 
+/**
+ * Returns the start time of the current GOP
+ *
+ * @param {Array} buffer
+ *        The current buffer of gop information
+ * @param {Player} player
+ *        The player instance
+ * @param {Double} mapping
+ *        Offset to map display time to stream presentation time
+ * @return {Double}
+ *         Time of current GOP
+ */
+export const currentGOPStart = (buffer, player, mapping) => {
+  if (!player || !buffer.length) {
+    return null;
+  }
+
+  // pts value for current time
+  const currentTimePts = Math.ceil((player.currentTime() - mapping) * 90000);
+
+  let i;
+
+  for (i = 0; i < buffer.length; i++) {
+    if (buffer[i].pts > currentTimePts) {
+      break;
+    }
+  }
+
+  if (i === 0 || buffer[i - 1].pts > currentTimePts) {
```

Comment:
I really wish I'd written that comment when I wrote the code. It's been so long that I don't remember :sweat:

Reply:
This is entirely based on the code, but it seemed like this was walking back one GOP to see if the currentTimePts was before anything in the buffer. I guess the assumption was that the buffer is sorted by pts and we could have overlapping buffer times? I don't know how accurate that is but maybe it jogs your memory

Reply:
> I guess the assumption was that the buffer is sorted by pts and we could have overlapping buffer times?

Hmmmmm. But we `break` at the first GOP that has a starting PTS greater than the current time, so I don't know how we'd ever end up in a situation where the previous GOP breaks that condition.

Honestly, this is probably just a programming error on my part. Since it's got the `i === 0` on the front, I wonder if maybe this was just bounds checking to see if any of our buffered GOPs includes the `currentTime`. In which case, I should probably just do that differently. So I'm going to code the change to be that, assuming that's what I was trying to do, because I really don't know what else it would be :\

Reply:
Heh, sounds good. I'm in favor of doing things that we remember/understand 😄